base/testfiles-lthooks/lthooks-002.tlg

```diff
@@ -31,6 +31,11 @@ Data structure for label rules:
 Handled code for kernel/filelist
 Handled code for kernel/warnings
 Update code for hook 'enddocument/end' on input line ...:
+Code labels for sorting:
+ kernel/hooks
+Data structure for label rules:
+ kernel/hooks = 0 -> 
+Handled code for kernel/hooks
 Update code for hook 'shipout/before' on input line ...:
```

Comment:
I don't like the idea of it being in a public hook either, but the last kernel hook we have in `\enddocument` is `\@kernel@after@enddocument@afterlastpage`, which is before reading the `.aux` in.  It could be used, or we could add another kernel hook after `enddocument/end` (`l3cctab` uses a similar checking at the end of the run).

Reply:
I would think that kernel hook is late enough. Reading the aux file should not affect that test and in some sense I think that check and error message should come before the warnings for labels  or the final file list. We could, as you say also add a kernel hooks after enddocument/end but for such test my feeling is the earlier place is better.

Reply:
That's a fair point.  I'll move to that hook.  I wonder if the `l3cctab` error should be moved earlier as well?

Reply:
interesting question. would it be possible that the l3cctab gets messed up when reading a (complete) aux file?
If not really then I think that should also go in the earlier kernel hook. Otherwise it should go in a kernel hook after enddocument/end but again my guess is that the earlier kernel hook is adequate.

Reply:
Only if someone does a stray `\cctab_begin:N` in the `.aux`, but then past this point (usually) no file is read so less stuff breaks (besides I can see no good reason for a catcode table change in the `.aux`).  I guess it can be reasonably assumed that that won't happen...  Same for the extra `\PushDefaultHookLabel`:  if it happens in the `.aux` it won't be caught, but there's no reason to use that in the `.aux` to begin with.  Besides, neither is a serious error: they are just there to help the programmer catch unbalanced code.  The earlier hook seems good to me for both.

Reply:
ok. then my proposel is to use this kernel hook here and also add an issue in l3 to move the l3cctab code also out of the user kernel and into that one.